From 69ba4137141ddd29397d22580c53262a911168c0 Mon Sep 17 00:00:00 2001
From: terminills <tdeters@gmail.com>
Date: Wed, 22 Jul 2026 06:55:34 -0500
Subject: [PATCH 264/638] Roll V8 for AROS TLS teardown fix

---
 v8 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8 b/v8
index d1700ca4e4..065bea9580 160000
--- a/v8
+++ b/v8
@@ -1 +1 @@
-Subproject commit d1700ca4e47bbcd381e4d35c8bcee4b69ed064e1
+Subproject commit 065bea9580071a39112bd8e9f990b291a5f34301
