From 9d334ac605f6a71b90a44fdff2db257a87fa3b28 Mon Sep 17 00:00:00 2001
From: terminills <tdeters@gmail.com>
Date: Fri, 7 Aug 2026 16:45:00 -0500
Subject: [PATCH 515/638] AROS: roll V8 mmap hint guard

---
 v8 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8 b/v8
index 9d79b0e5ef..4dcd8e2e86 160000
--- a/v8
+++ b/v8
@@ -1 +1 @@
-Subproject commit 9d79b0e5ef157648ad6334b75e81e2ddce5a9b22
+Subproject commit 4dcd8e2e8656de1698478e4991555ed58ccd2d7c
