From 0afe34fc8b2baeb469aea461502fc01c5e084789 Mon Sep 17 00:00:00 2001
From: terminills <tdeters@gmail.com>
Date: Sat, 8 Aug 2026 03:45:11 -0500
Subject: [PATCH 530/638] Roll AROS V8 allocator guard

---
 v8 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8 b/v8
index 4dcd8e2e86..c1f34904cc 160000
--- a/v8
+++ b/v8
@@ -1 +1 @@
-Subproject commit 4dcd8e2e8656de1698478e4991555ed58ccd2d7c
+Subproject commit c1f34904ccee1f14e8e7ec680c50be39cc911ae7
